src/services/WolfDen.Application/Requests/Queries/LeaveManagement/LeaveRequests/GetLeaveRequestHistory/GetLeaveRequestHistoryQueryHandler.cs

```diff
@@ -1,48 +1,55 @@
-﻿using MediatR;
+﻿using LanguageExt;
+using MediatR;
 using Microsoft.EntityFrameworkCore;
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
 using WolfDen.Application.DTOs.LeaveManagement;
-using WolfDen.Domain.Entity;
 using WolfDen.Infrastructure.Data;
 
 namespace WolfDen.Application.Requests.Queries.LeaveManagement.LeaveRequests.GetLeaveRequestHistory
 {
-    public class GetLeaveRequestHistoryQueryHandler : IRequestHandler<GetLeaveRequestHistoryQuery, List<LeaveRequestDto>>
+    public class GetLeaveRequestHistoryQueryHandler(WolfDenContext context) : IRequestHandler<GetLeaveRequestHistoryQuery, LeaveRequestHistoryResponseDto>
     {
-        private readonly WolfDenContext _context;
+        private readonly WolfDenContext _context = context;
 
-        public GetLeaveRequestHistoryQueryHandler(WolfDenContext context)
+        public async Task<LeaveRequestHistoryResponseDto> Handle(GetLeaveRequestHistoryQuery request, CancellationToken cancellationToken)
         {
-            _context = context;
-        }
+            int pageNumber = request.PageNumber > 0 ? request.PageNumber : 0;
+            int pageSize = request.PageSize > 0 ? request.PageSize : 1;
 
-        public async Task<List<LeaveRequestDto>> Handle(GetLeaveRequestHistoryQuery request, CancellationToken cancellationToken)
-        {
-            List<LeaveRequest> leaveRequests = await _context.LeaveRequests
-                  .Where(x => x.EmployeeId.Equals(request.RequestId))
-                 .Include(x => x.LeaveType).ToListAsync(cancellationToken);
+            int totalCount = await _context.LeaveRequests
+                                    .Where(x => x.EmployeeId.Equals(request.RequestId))
+                                    .CountAsync(cancellationToken);
+
+            int totalPages = (int)Math.Ceiling(totalCount / (double)pageSize);
 
-            List<LeaveRequestDto> leaveRequestList=new List<LeaveRequestDto>(); 
+            List<LeaveRequestDto> leaveRequestList = await _context.LeaveRequests
+            .Where(x => x.EmployeeId.Equals(request.RequestId))
+            .Include(x => x.LeaveType)
+            .Join(
```

Comment:
Why can't we achieve this result without using Join

Reply:
Updated it using '.include' employee